### ISSUE TITLE
Stabilization Scenario at 450 PPM resulting in errors

### ISSUE THREAD

**Issue by djtrainwreckx** [JGCRI/gcam-core#8]
Hi, I'm trying to run GCAM 4.3 scenario that stabilizes CO2 concentration at 450 PPM

the following text results and the final output is just the reference scenario

![issues with stabilization at 450 ppm](https://cloud.githubusercontent.com/assets/15677829/19454963/2e8a3340-9489-11e6-8dd8-9c2a4906dc72.png)

I am using this code for 'stabilization_450_PPM' policy (this one has look forward of 5, the problem happened with 1 too) and config file attached as txt (xml files can't be uploaded)

[450_PPM_stabilization.txt](https://github.com/JGCRI/gcam-core/files/534687/450_PPM_stabilization.txt)

[configuration.txt](https://github.com/JGCRI/gcam-core/files/534686/configuration.txt)

![results never go past reference](https://cloud.githubusercontent.com/assets/15677829/19455350/c5470fbe-948a-11e6-8b06-fee03e828a19.png)

-Osman

**Comment by djtrainwreckx** [comment]
Same thing occurred when I tried to stabilize at 725 PPM, BAU is 748 PPM

![issues with stabilization at 750 ppm](https://cloud.githubusercontent.com/assets/15677829/19456137/1acfb1cc-948e-11e6-840d-53dc481bf557.png)

**Comment by rplzzz** [comment]
@djtrainwreckx There are a couple of things going on here:

1. You have set your target type to "stabilization".  This requests an "emissions stabilization" target, which is not what you had intended.  You want a "concentration" target with the stabilization option set (which it is).

2. There is a bug in the `HectorModel::getEmissions` method, which causes it to reject requests for emissions in the final year.  That's the source of the "invalid year" error message in your runs, and it causes the run to bail early.  The other target types don't use the emissions directly (they call `getConcentration`, `getTotalForcing`, or `getTemperature`), so if you switch to one of them, you won't see those errors.

3. You set your `target-tolerance` parameter to be exactly 0.  That's not going to work.  You should stick with the default of 0.01.  You could go lower if you want, say to 0.001, but then you're doing a lot of additional computing for no real benefit.

So, to summarize, there are a couple of small bugs we need to fix here, but you can get the runs you want to do working by selecting the correct target type ("concentration" in your case) and raising `target-tolerance` to something greater than zero.

**Comment by djtrainwreckx** [comment]
Thanks for the reply, rplzzz!

I tried the unaltered forcing_4p5 policy with no alterations to it and the config only changing 'find path' to 1

![gcam 4 3 not working on default forcing 4 5 policy](https://cloud.githubusercontent.com/assets/15677829/19499069/5ab67b3e-9562-11e6-828b-aa2512b81e05.png)

![forcing_4p5 does look like it works](https://cloud.githubusercontent.com/assets/15677829/19499137/d2bd6a48-9562-11e6-9448-9827e6420c76.png)


I will try your recommendations, especially using 'concentration' instead of 'stabilization'

-Osman

**Comment by rplzzz** [comment]
I believe that what's happening in your run is that the stabilization algorithm is having a difficult time dealing with the lags in the energy and climate systems.  You can see some of this in the CO<sub>2</sub> prices, where the price drops immediately after the onset of the stabilization (as contrasted with mitigation) regime, as the model tries to slow down the pace of emissions reductions.  This turns out to be an overcorrection, and so the carbon price has to rise dramatically in 2090 to keep us on track, and _that_ turns out to be an overcorrection to the overcorrection, so the price has to drop again in 2095.  By 2100 there is enough mitigation baked into the system that even a zero carbon price causes us to undershoot the target, and that's the cause of the error message.  The resulting scenario is valid, but it doesn't exactly meet the goal of stabilizing the forcing at 4.5 W/m<sup>2</sup>.  Forcing ticks slightly above 4.5 in 2090 and then well below 4.5 by 2100.

**Comment by djtrainwreckx** [comment]
rplzz, these kinds of problems popped up for me since I started using GCAM (4.1?). 

By increasing the forward look, I think the GCAM solver usually found the more realistic solution

![forward look solves crazy swings](https://cloud.githubusercontent.com/assets/15677829/19533907/b0a42764-960f-11e6-9dbe-4ac9ca5ca29f.png)

[4p5 with forward look.txt](https://github.com/JGCRI/gcam-core/files/540001/4p5.with.forward.look.txt)

**Comment by rplzzz** [comment]
Good stuff.  The stabilization part of the target finding hasn't been looked at in while.  We've got some ideas on how to improve it, so keep an eye out.
